### DISCUSSION TITLE
Serve jekyll site inside code-server container and base url is not proxied

### DISCUSSION THREAD

**Discussion by CoderSherlock** [coder/code-server#6498] [Q&A]
Hi folks,
I tried to serve a static website inside code-server container. When clicking the open in browser button, it redirects to `{code-server-ip}/proxy/4000`. However, only the index.html is available, all other links/css/images are based on the wrong baseurl, such as `{code-server-ip}/about.html}` (expect `{code-server-ip}/proxy/4000/about.html` I guess). 

![image](https://github.com/coder/code-server/assets/4081940/2cea581e-2f85-4358-af87-e8179f357a0a)

```
GET 192.168.0.10/proxy/4000 ...OK
GET 192.168.0.10/assets/css/main.css net::ERR_ABORTED 404 (Not Found)
```

Do we have a good way to proxy all resources? Is there an existing solution to this case?

---
Ref: similar question from another static website generator [Code-server proxy and properly setting up hugo server](https://discourse.gohugo.io/t/code-server-proxy-and-properly-setting-up-hugo-server/42819).

**Comment by code-asher** [comment]
It will depend on the static generator.  But some general thoughts:

1. If you can, use the subdomain-based proxy as it avoids all these issues.
2. You might need to use `/absproxy` instead of `/proxy`.  It all depends on the application being proxied.  Some of them need the base path in the URL (`/absproxy` does this), and some need it stripped out (`/proxy` does this, if you are familiar with NGINX it is the same as a trailing slash in `proxy_pass`).
3. The static generator might have a way to specify a base URL (`/proxy/4000` for example) or, more ideally, a way to make URLs relative.
